homeassistant/components/madvr/remote.py

```diff
@@ -0,0 +1,176 @@
+
+    @property
+    def should_poll(self) -> bool:
+        """Poll."""
+        return False
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return extra state attributes."""
+        # Useful for making sensors
+        return self.madvr_client.msg_dict
+
+    async def handle_queue(self):
+        """Handle command queue."""
+        while True:
+            await self.connection_event.wait()
+            while (
+                not self.command_queue.empty()
+                and not self.stop_processing_commands.is_set()
+            ):
+                command = await self.command_queue.get()
+                _LOGGER.debug("sending queue command %s", command)
+                try:
+                    await self.madvr_client.send_command(command)
+                except ConnectionResetError:
+                    _LOGGER.warning("Envy was turned off manually")
+                    await self.async_turn_off()
+                except AttributeError:
+                    _LOGGER.warning("Issue sending command from queue")
+                except RetryExceededError:
+                    _LOGGER.warning("Retry exceeded for command %s", command)
+                except OSError as err:
+                    _LOGGER.error("Unexpected error when sending command: %s", err)
+                finally:
+                    self.command_queue.task_done()
+
+            if self.stop_processing_commands.is_set():
+                await self.clear_queue()
+                _LOGGER.debug("Stopped processing commands")
+                break
+
+            await asyncio.sleep(0.1)
+
+    async def clear_queue(self):
+        """Clear queue."""
+        self.command_queue = asyncio.Queue()
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off the device."""
+        _LOGGER.debug("Turning off")
+        self.stop_processing_commands.set()
+        await self.clear_queue()
+        await self.madvr_client.power_off()
+        self._attr_is_on = False
+        _LOGGER.debug("self._state is now: %s", self._attr_is_on)
+
+        # Fire power state change event
+        event_data = {
+            "device_id": self.entry_id,
+            "type": "power_state_changed",
+            "new_state": "off",
+            "old_state": "on",
+        }
+        self.hass.bus.async_fire(f"{DOMAIN}_event", event_data)
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on the device."""
+        _LOGGER.debug("Turning on device")
+        await self.madvr_client.power_on()
+        self.stop_processing_commands.clear()
+        _LOGGER.debug("Firing event to turn on")
+        # Fire power state change event
+        event_data = {
+            "device_id": self.entry_id,
+            "type": "power_state_changed",
+            "new_state": "on",
+            "old_state": "off",
+        }
+        self.hass.bus.async_fire(f"{DOMAIN}_event", event_data)
```

Comment:
What is this used for?

Reply:
that should have been removed, will remove this